### PR TITLE
Set classes

### PR DESCRIPTION
**Pull request by willmcgugan**
Adds the ability to change / clear classes.

- Adds a setter for classes, which may be intuitive to some.
- Also adds explicit `set_classes`

Fixes https://github.com/Textualize/textual/issues/1081

### CODE REVIEW

File: src/textual/dom.py [hunk 1]
```diff
@@ -83,6 +83,26 @@ class NoScreen(DOMError):
     pass
 
 
+class _ClassesDescriptor:
```
Comment (willmcgugan) [inline src/textual/dom.py:86]: Mypy doesn't like it if you have a property where the setter and getter have different types. However, it is happy with a descriptor.

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +40,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Added Screen.ModalScreen which prevents App from handling bindings. https://github.com/Textualize/textual/pull/2139
 - Added TEXTUAL_LOG env var which should be a path that Textual will write verbose logs to (textual devtools is generally preferred) https://github.com/Textualize/textual/pull/2148
 - Added textual.logging.TextualHandler logging handler
+- Added Query.set_classes DOMNode.set_classes and `classes` setter for Widget https://github.com/Textualize/textual/issues/1081
```
Comment (davep) [inline CHANGELOG.md:43]:
```suggestion
- Added Query.set_classes, DOMNode.set_classes, and `classes` setter for Widget https://github.com/Textualize/textual/issues/1081
```

File: src/textual/dom.py [hunk 3]
```diff
@@ -888,6 +905,15 @@ def set_class(self, add: bool, *class_names: str) -> None:
         else:
             self.remove_class(*class_names)
 
+    def set_classes(self, classes: str | Iterable[str]) -> None:
```
Comment (davep) [inline src/textual/dom.py:908]: Perhaps this should return `self` so folk can chain calls?

**Review by davep — APPROVED** [review]
Text tweak and an idle question; but nothing of consequence form me. This looks super handy!

File: src/textual/dom.py (hunk 3 above)
Comment (willmcgugan) [reply src/textual/dom.py:908]: It probably should. We don't have that on the other classes either.

**Review by rodrigogiraoserrao — APPROVED** [review]

**Pull request merged**